django/db/migrations/loader.py

```diff
@@ -316,6 +319,33 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
 
+    def check_consistent_history(self, connection):
+        """
+        Checks that all migrations are applied after their dependencies. Raises
+        InconsistentMigrationHistory exception when migration in applied before
+        it's dependency
+        """
+        recorder = MigrationRecorder(connection)
+        applied = recorder.applied_migrations()
+        for migration in applied:
+            # If we don't know anything about this migration we skip it
+            if migration not in self.graph.nodes:
+                continue
+            unapplied_parents = [
+                x.key for x in self.graph.node_map[migration].parents
+                if x not in applied and settings.MIGRATION_MODULES.get(x[0])
```

Comment:
@MarkusH Thanks for your proposal, but this condition:
```
x[0] in settings.MIGRATION_MODULES and settings.MIGRATION_MODULES[x[0]] is not None
```
is basically equivalent to `settings.MIGRATION_MODULES.get(x[0]) is not None`. And as @timgraham wrote we want detection to work regardless if that setting is defined.

I'll rewrite it into the loop once this issue will be resolved.